spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -1315,6 +1346,64 @@ def get_incoming_projections_from(self, source_vertex):
         """
         return self.__incoming_projections[source_vertex]
 
+    @property
+    def incoming_poisson_projections(self):
+        """ The projections that target this population vertex which
+            originate from a Poisson source
+
+        :rtype: iterable(~spynnaker.pyNN.models.projection.Projection)
+        """
+        return self.__incoming_poisson_projections
+
+    @property
+    def pop_seed(self):
+        """ The seed to use for the population overall
+
+        :return: A list of 4 integers
+        :rtype: list(int)
+        """
+        return self.__pop_seed
+
+    def core_seed(self, vertex_slice):
+        """ The seed to use for a core
+
+        :param Slice vertex_slice: The machine vertex that the seed is for
+
+        :return: A list of 4 integers
+        :rtype: list(int)
+        """
+        if vertex_slice not in self.__core_seeds:
+            self.__core_seeds[vertex_slice] = create_mars_kiss_seeds(
+                self.__rng)
+        return self.__core_seeds[vertex_slice]
+
+    def copy_initial_state_variables(self, vertex_slice):
+        """ Copies the state variables into the initial state variables
+
+        :param Slice vertex_slice: The slice to copy now
+        """
+        for key in self.__state_variables.keys():
+            value = self.__state_variables[key][vertex_slice.as_slice]
+            self.__initial_state_variables[key].set_value_by_slice(
+                vertex_slice.lo_atom, vertex_slice.hi_atom + 1, value)
+        # This is called during reading of initial values, so we don't
+        # need to do it again
+        self.__read_initial_values = False
+        self.__have_read_initial_values = True
+
+    @property
+    def read_initial_values(self):
+        """ Determine if initial values need to be stored
+
+        :rtype: bool
+        """
+        return self.__read_initial_values
+
+    def __tell_neuron_vertices_to_regenerate(self):
+        for vertex in self.machine_vertices:
```

Comment:
SpynnakerDataView.set_requires_data_generation()

Reply:
Reload is enough for neurons